pygfx/renderers/wgpu/_shaderbase.py

```diff
@@ -262,14 +262,25 @@ def __setitem__(self, key, value):
     def __getitem__(self, key):
         return self.kwargs[key]
 
+    @property
     def hash(self):
         """A hash of the current state of the shader. If the hash changed,
         it's likely that the shader changed.
         """
-        h = hashlib.sha1()
-        h.update(repr(self.kwargs).encode())
-        h.update(self.code_definitions().encode())
-        return h.hexdigest()
+        # The full name of this shader class.
+        fullname = self.__class__.__module__ + "." + self.__class__.__name__
+
+        # Does it look like its builtin (into a lib)? This excludes e.g. __main__.CustomShader
+        is_builtin = fullname.count(".") >= 2
```

Comment:
In general it should be safe to use the name. The only case where it breaks down is when the class is re-defined, as can happen in an interactive session. This is easily checked by counting dots. It's not 100% bulletproof, since in theory a module can be reloaded, but that's rare and bad practice anyway.

I will change the comments here to clarify.